spec/javascripts/components/intervention-spec.js

```diff
@@ -0,0 +1,91 @@
+
+  var container
+
+  beforeEach(function () {
+    container = document.createElement('div')
+    container.innerHTML =
+      '<section class="gem-c-intervention" data-module="intervention" data-intervention-name="test-intervention-name"><a class="govuk-link gem-c-intervention__dismiss-link">Dismiss</a></section>'
+
+    document.body.appendChild(container)
+    var element = document.querySelector('[data-module="intervention"]')
+    new GOVUK.Modules.Intervention(element).init()
+
+    window.GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+  })
+
+  afterEach(function () {
+    GOVUK.deleteCookie('intervention_campaign')
+    document.body.removeChild(container)
+  })
+
+  describe('close banner', function () {
+    it('should hide intervention banner', function () {
+      var banner = document.querySelector('.gem-c-intervention')
+      var close = document.querySelector('.gem-c-intervention__dismiss-link')
+      expect(banner).toBeVisible()
+
+      close.click()
+
+      expect(banner).toBeHidden()
+    })
+
+    it('sets a cookie value', function () {
+      spyOn(GOVUK, 'cookie').and.callThrough()
+      var close = document.querySelector('.gem-c-intervention__dismiss-link')
+      close.click()
+
+      expect(GOVUK.cookie).toHaveBeenCalled()
+      var bannerCookie = GOVUK.cookie('intervention_campaign')
+      expect(bannerCookie).toEqual('test-intervention-name')
+    })
+  })
+
+  describe('cookies are already set', function () {
+    it('does not display the banner', function () {
+      GOVUK.setCookie('intervention_campaign', 'test-intervention-name', { days: 1 })
+      var element = document.querySelector('[data-module="intervention"]')
+      new GOVUK.Modules.Intervention(element).init()
+
+      var banner = document.querySelector('.gem-c-intervention')
+
+      expect(banner).toBeHidden()
+    })
+  })
+})
+
+describe('another concurrent banner campaign', function () {
+  'use strict'
+
+  var container
+
+  beforeEach(function () {
+    GOVUK.setCookie('intervention_campaign', 'existing-value', { days: 1 })
+    window.GOVUK.setCookie('cookies_policy', '{"essential":true,"settings":true,"usage":true,"campaigns":true}')
+
+    container = document.createElement('div')
+    container.innerHTML =
+      '<section class="gem-c-intervention" data-module="intervention" data-intervention-name="test-intervention-name"><a class="govuk-link gem-c-intervention__dismiss-link">Dismiss</a></section>'
+
+    document.body.appendChild(container)
+    var element = document.querySelector('[data-module="intervention"]')
+    new GOVUK.Modules.Intervention(element).init()
+  })
+
+  it('appends campaign value to the cookie', function () {
+    var close = document.querySelector('.gem-c-intervention__dismiss-link')
+    close.click()
+
+    var updatedCookie = GOVUK.cookie('intervention_campaign')
+    expect(updatedCookie).toEqual('existing-value,test-intervention-name')
```

Comment:
Is there a way to check that the expiry date of the cookie has been extended? My understanding of the code is that since we're using a single cookie, when there is an existing campaign the new one is appended and the expiry date extended to 30 days from now. 

Currently I think the behaviour of the code is either:

- 1st Jan: user hides intervention banner on campaign A, expires in 30 days
- 29th Jan: user hides intervention banner on campaign B, uses existing cookie, extends cookie expiry
- campaign A and B now have the banner reappear 30 days after the 29th Jan

One campaign having an effect on all campaigns doesn't feel ideal, but better than the alternative:

- 1st Jan: user hides intervention banner on campaign A, expires in 30 days
- 29th Jan: user hides intervention banner on campaign B, uses existing cookie, doesn't change expiry date
- campaign A and B have the banner reappear 30 days after the original cookie, i.e. one day from the 29th Jan

If neither of those is ideal there might be a way of storing the individual campaign expiry date inside the cookie, then setting the cookie expiry to the latest value. But that way further complexity lies.

Reply:
I think scenario 1 is what I should aim for...

Reply:
Agreed, it's far simpler. Would be good to have a test to check that it's doing that, if possible.

Reply:
As agreed from our call, I won't add a test for this as it's impossible to affect and read the cookie expiration values in Jasmine. Plus it feels like testing for browser behaviour at this point. The expected behaviour of the cookie expiry extending when clicking on another campaign banner was checked during the call.